core/CacheId.php

```diff
@@ -26,4 +26,16 @@ public static function pluginAware($cacheId)
 
         return $cacheId;
     }
+
+    public static function siteAware($cacheId, array $idSites = null)
+    {
+        if ($idSites === null) {
+            $idSitesStr = Common::getRequestVar('idSite', false);
```

Comment:
can we cast `idSite` to int? Just to be sure it doesn't in the end include like `../../../etc/passwd` or something funny? 

I was thinking to otherwise use `Site::getIdSitesFromIdSitesString(Common::getRequestVar('idSite', false))` but this could potentially result in lots of different caches depending on what `idSite=all` translates to. 

I was also wondering if there could be any security issue when `idSite=all` and the cache was generated by a super user, but then later accessed by an admin who was only access to some sites... but I believe this should not be a problem and the same case as it is currently anyway. 
Maybe we can check `idSiteStr` is numeric, and if not, use `0` or so?

Reply:
What if it is 'all'? Will 0 work in that case? I at first wanted to call the SitesManager API, but thought that would be too much.

Reply:
`0` will work. Calling sitesManager (eg done by `Site::getIdSitesFromIdSitesString`) would result in heaps of different caches which we maybe want to avoid. (As each user may have access to a different set of sites)